qt/scientific_interfaces/Indirect/IndirectFitPlotPresenter.cpp

```diff
@@ -207,9 +213,9 @@ void IndirectFitPlotPresenter::updateSelectedDataName() {
 void IndirectFitPlotPresenter::updateDataSelection() {
   MantidQt::API::SignalBlocker blocker(m_view);
   m_view->clearDataSelection();
-  for (auto i = 0u; i < m_model->numberOfWorkspaces(); ++i)
+  for (DatasetIndex i{0}; i < m_model->numberOfWorkspaces(); ++i)
```

Comment:
Swap to `int i`

Reply:
I think it's more complicated to switch to int i here than use the DatasetIndex as both the comparitor in the for loop and the functions in the body expect DataSet indexes